src/components/ChatEntry.js

```diff
@@ -1,22 +1,48 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const [isLiked, setIsLiked] = useState(props['liked']);
+
+  // console.log(props)
+
+  const likeButtonClick = () => {
+    setIsLiked( !isLiked );
+    const messageData= {
+      id: props['id'],
+      body: props['body'],
+      sender: props['sender'],
+      timeStamp: props['timeStamp'],
+      liked: !isLiked,
+    };
+    props.updatedMessage(messageData)
```

Comment:
By having the ChatEntry component create the updated message, the component now needs to have extra knowledge of the format App.js uses to hold the data. To keep this presentational component as simple as possible, I suggest structuring the `updatedMessage` callback to take in an ID of the message to update, and have App.js handle creating the new message in the correct format for the `newChatMessages` state.